BlobStorage/Advanced.cs

```diff
@@ -1377,50 +1137,41 @@ private static async Task GetExistingBlobReferenceAsync(CloudBlobContainer conta
                     throw;
                 }
             }
-
             Console.WriteLine();
         }
 
         /// <summary>
         /// Creates the specified number of sequentially named block blobs, in a flat structure.
         /// </summary>
-        /// <param name="container">A CloudBlobContainer object.</param>
+        /// <param name="container">A BlobContainerClient object.</param>
         /// <param name="numberOfBlobs">The number of blobs to create.</param>
         /// <returns>A Task object.</returns>
-        private static async Task CreateSequentiallyNamedBlockBlobsAsync(CloudBlobContainer container, int numberOfBlobs)
+        private static async Task CreateSequentiallyNamedBlockBlobsAsync(BlobContainerClient container, int numberOfBlobs)
         {
             try
             {
-                CloudBlockBlob blob;
+                BlobClient blob;
                 string blobName = string.Empty;
-                MemoryStream msWrite;
 
                 for (int i = 1; i <= numberOfBlobs; i++)
                 {
                     // Format string for blob name.
                     blobName = i.ToString("00000") + ".txt";
 
                     // Get a reference to the blob.
-                    blob = container.GetBlockBlobReference(blobName);
+                    blob = container.GetBlobClient(blobName);
 
-                    // Set a property on the blob.
-                    blob.Properties.ContentType = "text/html";
+                    Dictionary<string, string> metadata = new Dictionary<string, string>();
 
+                    await blob.UploadAsync(BinaryData.FromString($"This is blob {blobName}"));
                     // Set some metadata on the blob.
-                    blob.Metadata.Add("DateCreated", DateTime.UtcNow.ToLongDateString());
-                    blob.Metadata.Add("TimeCreated", DateTime.UtcNow.ToLongTimeString());
+                    metadata.Add("DateCreated", DateTime.UtcNow.ToLongDateString());
+                    metadata.Add("TimeCreated", DateTime.UtcNow.ToLongTimeString());
 
-                    // Write the name of the blob to its contents as well.
-                    msWrite = new MemoryStream(Encoding.UTF8.GetBytes("This is blob " + blobName + "."));
-                    msWrite.Position = 0;
-                    using (msWrite)
-                    {
-                        // Uploading the blob sets the properties and metadata on the new blob.
-                        await blob.UploadFromStreamAsync(msWrite);
-                    }
+                    await blob.SetMetadataAsync(metadata);
```

Comment:
Is there a reason we don't set the metadata on the upload itself? That would be good to demonstrate in a sample.

Reply:
@jaschrep-msft - @tg-msft let me use the UploadAsync that takes a BinaryData.FromString("") instead of all the Stream code in this sample, so I don't set the metadata on the upload itself.

Reply:
You can still set the metadata on the upload itself. The API doesn't prevent you.
```csharp
Dictionary<string, string> metadata = new Dictionary<string, string>();
metadata.Add("DateCreated", DateTime.UtcNow.ToLongDateString());
metadata.Add("TimeCreated", DateTime.UtcNow.ToLongTimeString());
await blob.UploadAsync(BinaryData.FromString("my data"), new BlobUploadOptions
{
    Metadata = metadata
});
```